src/commands/guess.ts

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable max-len */
+// pls suggestions, i want make some kind of json file containing all the replies (it would make the translation easier)
+import dayjs from 'dayjs';
+import { CommandInteraction, MessageActionRow, MessageButton } from 'discord.js';
+import { RESTPostAPIChatInputApplicationCommandsJSONBody } from 'discord-api-types/v10';
+import { Command } from '../interfaces/Command';
+import { checkOrRegisterUser, getDay, getWord, played } from '../database';
+import { runAtMidnight } from '../utils/runner';
+import { awaitMessage } from '../utils/msgCollector';
+import { isValid } from '../utils/checkWord';
+import { toDefault, toEmoji } from '../utils/converters';
+import { plataform } from '../utils/plataform';
+import { hasPermissions } from '../utils/permissions';
+import { square } from '../utils/emotes.json';
+
+let usersTries: Record<string, Record<string, string | string[]>> = {};
+let activeGames: string[] = [];
+
+runAtMidnight(() => {
+	usersTries = {};
+	activeGames = [];
+});
+
+export default class Guess implements Command {
+	commandStructure: RESTPostAPIChatInputApplicationCommandsJSONBody = {
+		'name': 'adivinhar',
+		'description': 'Tente adivinhar a palavra do dia!',
+	};
+
+	dev = true;
+
+	async execute(interaction: CommandInteraction) {
+		const { user, channel } = interaction;
+
+		if (!hasPermissions(interaction, ['MANAGE_MESSAGES', 'USE_EXTERNAL_EMOJIS'])) return;
+
+		if (await checkOrRegisterUser(user.id)) {
+			const t = dayjs().tz('America/Sao_Paulo').endOf('day').unix();
+			interaction.reply({
+				'content': `Você já jogou hoje!\nTempo restante até a próxima palavra: <t:${t}:R>`,
+				'ephemeral': true,
+			});
+			return;
+		}
+
+
+		const gameMessage: Record<number, string> = Object.create(null);
+		for (let i = 0; i < 6; i++) gameMessage[i + 1] = square.gray.repeat(5);
+
+		function table(): string {
+			const u = usersTries[user.id];
+			if (u) {
+				for (let i = 0; i < 6; i++) {
+					u.attempts[i] ? gameMessage[i + 1] = u.attempts[i] : gameMessage[i + 1] = square.gray.repeat(5);
+				}
+			}
+
+			return Object.values(gameMessage).map(line => line).join('\n');
+		}
+
+
+		const playingUser: Record<string, string | string[]> = { id: user.id, attempts: [] };
+		const correctWord = await getWord();
+
+		let i = 0;
+		if (activeGames.includes(user.id)) {
+			await interaction.reply('Você já está jogando!\nSe a mensagem não aparece mais, mande `cancelar` no canal e tente novamente.');
+			return;
+		} else {
+			activeGames.push(user.id);
+
+			await interaction.reply({
+				'content': `Adivinhe o **PALAVRECO** de hoje! 👀\n\n${table()}\nPara cancelar o jogo, digite \`cancelar\``,
+				'ephemeral': true,
+			});
+
+			const u = usersTries[user.id];
+			u ? i = u.attempts.length : usersTries[user.id] = playingUser;
```

Comment:
I found this line a little bit confusing... I think it's worth a refactoring.

Reply:
I just simplified this messy code :P
![image](https://user-images.githubusercontent.com/79790819/165309932-2435d852-2d36-4224-928a-adaa796b5074.png)

Do you have any suggestions?

Reply:
I don't think that's messy lol

Only the unecessary const